numpy/core/_add_newdocs.py

```diff
@@ -1350,6 +1350,27 @@
     >>> np.arange(3,7,2)
     array([3, 5])
 
+    .. WARNING::
+        Using non-integer steps in ``np.arange`` is discouraged and may lead to
+        confusing and non intuitive results.
+        For example, using ``dtype=np.int`` with fractional steps will cause
+        this behaviour:
+
+        >>> np.arange(-0.09, 10.0, 1.9, dtype=int)
+        array([0, 1, 2, 3, 4, 5])
+        >>> np.arange(-0.01, 10.0, 1.009, dtype=int)
+        array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
```

Comment:
```suggestion
        >>> np.arange(0, 5, 0.5, dtype=int)
        array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
        >>> np.arange(-3, 3, 0.5, dtype=int)
        array([-3, -2, -1,  0,  1,  2,  3,  4,  5,  6,  7,  8])
```
I think this could be improved with easier-to-grok examples. I've left some suggestions but feel free to come up with others.